### ISSUE TITLE
Svelte's tooling ignores TypeScript code from previous `preprocessors`

### ISSUE THREAD

**Issue by ivanhofer** [sveltejs/language-tools#1567]
### Description

Speaking of the VS Code extension and the `svelte-check` command, when I'm adding my own preprocessor that modifies the script tag to include additional `TypeScript` syntax, that part gets ignored by the tooling.

I tried `svelte-check` and added some debug logs to see if the custom preprocessor get's loaded. The logs show up, but If I'm adding TypeScript syntax, the tooling ignores it. 

Not sure what exactly happens, but it seems that the last preprocessor doesn't use the full output from the previous preprocessors.

### Proposed solution

The VS Code extension and `svelte-check` should take the input from the previous preprocessors.

### Alternatives

Leave it like it is and don't allow custom use-cases.

### Additional Information, eg. Screenshots

Here is my _`svelte.config.js`_ file
```ts
import preprocess from 'svelte-preprocess'

/** @type {import('@sveltejs/kit').Config} */
const config = {
	// Consult https://github.com/sveltejs/svelte-preprocess
	// for more information about preprocessors
	preprocess: [
		{
			script: ({ content, attributes, filename }) => {
				if (filename.includes('/src/routes/') && attributes.lang === 'ts' && attributes.context === 'module') {
					console.log(`1. the custom preprocessor get's used for '${filename}'`)
					const file = filename.substring(filename.lastIndexOf('/') + 1)
					const path = file.substring(0, file.indexOf('.'))
					const typePath = `./__types/${path}`

					console.log(`2. adding type import for '${filename}'`)
					return {
						code: `
import type { Load } from '${typePath}';
console.log('this gets logged');
${content}`,
					}
				}
			},
		},
		{
			script: ({ markup, content, attributes, filename }) => {
				if (filename.includes('/src/routes/') && attributes.lang === 'ts' && attributes.context === 'module') {
					console.log(`3. the resulting output for '${filename}' is:
${content}
`)
				}
			},
		},
		preprocess(),
	],
}

export default config
```

The first preprocessor will add the generated type imports for the `Load` function.
The second preprocessor checks if the first preprocessor modified the source code
The third preprocessor is the usual `svelte-preprocess` that ccan handle `TypeScript` syntax

Here is my custom route _`routes/test/[id]/summary`_:

```svelte
<script context="module" lang="ts">
	export const load: Load = ({ params }) => {
		return { props: { id: params.id } }
	}
</script>

<script lang="ts">
	export let id: number
</script>

hello {id}
```

When running `svelte-check` I get following error:
![image](https://user-images.githubusercontent.com/21335119/180602305-d1ea774e-b59c-4f62-92e8-a87dc38d6c4b.png)

As you can see my custom preprocessor gets executed, but in the end I still see an error because the type import for `Load` is missing.

**Comment by jasonlyu123** [comment]
This is primarily a duplicate of https://github.com/sveltejs/language-tools/issues/339  although that one is specific about template preprocess the limitation we have still applied. The type check step doesn't use the preprocess config. 

It could also be argued that even if we were to fix that issue. The script preprocesses would probably still be ignored. There's currently no way to tell if the preprocess is intended to be used for type checks.

**Comment by ivanhofer** [comment]
Thanks, I now understand the problem better.
So the propreocessor structure would need to be updated to be able to distinguish between "regular" preprocessors and preprocessors that make TypeScript changes? And ideally this type of preprocessor is performed in a synchronous way.

The issue you have linked (#339) is about making an asynchronous process synchronous. So I think those are two different problems that can be solved independently, right?

**Comment by twag123** [comment]
I got this configuration to work by having a separate `playwright-ct.config.ts`:
```javascript
import { defineConfig, devices } from '@playwright/experimental-ct-svelte';
import { svelte, vitePreprocess } from '@sveltejs/vite-plugin-svelte';
import { resolve } from 'node:path';

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
	testDir: './',
	/* The base directory, relative to the config file, for snapshot files created with toMatchSnapshot and toHaveScreenshot. */
	snapshotDir: './__snapshots__',
	/* Maximum time one test can run for. */
	timeout: 10 * 1000,
	/* Run tests in files in parallel */
	fullyParallel: true,
	/* Fail the build on CI if you accidentally left test.only in the source code. */
	forbidOnly: !!process.env.CI,
	/* Retry on CI only */
	retries: process.env.CI ? 2 : 0,
	/* Opt out of parallel tests on CI. */
	workers: process.env.CI ? 1 : undefined,
	/* Reporter to use. See https://playwright.dev/docs/test-reporters */
	reporter: 'html',
	/* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
	use: {
		/* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
		trace: 'on-first-retry',

		/* Port to use for Playwright component endpoint. */
		ctPort: 3100,

		ctViteConfig: {
			plugins: [svelte({ preprocess: vitePreprocess() })],
			resolve: {
				alias: {
					$lib: resolve('src/lib')
				}
			}
		}
	},

	/* Configure projects for major browsers */
	projects: [
		{
			name: 'chromium',
			use: { ...devices['Desktop Chrome'] }
		},
		{
			name: 'firefox',
			use: { ...devices['Desktop Firefox'] }
		},
		{
			name: 'webkit',
			use: { ...devices['Desktop Safari'] }
		}
	]
});
```

Then here is my `playwright.config.ts` file:
```javascript
import type { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
	webServer: {
		command: 'npm run build && npm run preview',
		port: 4173
	},
	testDir: 'tests',
	testMatch: /(.+\.)?[a-z]*(Test|Spec)\.[jt]s/
};

export default config;
```
